### PR TITLE
deps!: Update multiaddr & multihash to 0.17.0

### PR DESCRIPTION
**Pull request by ppodolsky**
## Notes

Dependabot forgot to update multihash in https://github.com/libp2p/rust-libp2p/pull/3193/files, fixed this.

Depends-On: https://github.com/libp2p/rust-libp2p/pull/3203

### CODE REVIEW

**Comment by b5** [comment]
👋 libp2p team, would love to know if this is mergable. It's a blocker on https://github.com/n0-computer/iroh/pull/546 for us in iroh. Thanks!

**Review by thomaseizinger — APPROVED** [review]
LGTM, thank you!

I kicked CI.

**Comment by thomaseizinger** [comment]
The failing interop test should be fixed once https://github.com/libp2p/rust-libp2p/pull/3203 is merged.
The two failing semver tests are due to a bug in `cargo semver-checks`: https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994

@mxinden Will need to exercise his admin rights to merge this until we have that sorted.

**Comment by ppodolsky** [comment]
@thomaseizinger Hi! Could you push it forward? Is there anything I can do to help?

**Comment by thomaseizinger** [comment]
> @thomaseizinger Hi! Could you push it forward? Is there anything I can do to help?

Nope, I don't have the admin rights to merge a PR with failing builds. We need to merge and release https://github.com/libp2p/rust-libp2p/pull/3213 to fix the false-positive reports from `cargo semver-checks`.

**Comment by thomaseizinger** [comment]
I've approved it to go into the merge-queue but that won't happen until the build is green. Keep an eye on the conversation in #3213. Once that is resolved and released, you should be able to get this in by updating it with latest master to trigger CI again at which point it should automatically go in once it is green!

File: core/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,8 @@ futures-timer = "3"
 instant = "0.1.11"
 libsecp256k1 = { version = "0.7.0", optional = true }
 log = "0.4"
-multiaddr = { version = "0.16.0" }
-multihash = { version = "0.16", default-features = false, features = ["std", "multihash-impl", "identity", "sha2"] }
+multiaddr = { version = "0.17.0" }
```
Comment (mxinden) [inline core/Cargo.toml:24]: `multiaddr` is reexported and exposed on the public `libp2p-core` interface. Thus this is a breaking change and thereby requires version bumps across most of our crates. I can push a commit that does that in a bit.

**Review by mxinden — DISMISSED** [review]
Thanks for the patch @ppodolsky.

**Comment by mxinden** [comment]
@ppodolsky I seem to not have permissions to push to your branch. Can you please cherry pick https://github.com/mxinden/rust-libp2p/commit/72248a1064b89a7658c1a4f46032d944d4ea18f9 ?

@thomaseizinger any objections including this in the next release? In other words, any objections in doing a full breaking change for the next release?

**Comment by ppodolsky** [comment]
Done

**Comment by thomaseizinger** [comment]
> @thomaseizinger any objections including this in the next release? In other words, any objections in doing a full breaking change for the next release?

Nope, fine with me!

File: transports/webrtc/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# 0.5.0 [unreleased]
```
Comment (thomaseizinger) [inline transports/webrtc/CHANGELOG.md]:
```suggestion
# 0.4.0-alpha.2 [unreleased]
```

File: transports/tls/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# 0.2.0 [unreleased]
```
Comment (thomaseizinger) [inline transports/tls/CHANGELOG.md]:
```suggestion
# 0.1.0-alpha.2 [unreleased]
```

File: transports/tls/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
...
+
+- Update to `libp2p-core` `v0.39.0`.
+
 # 0.1.0
```
Comment (thomaseizinger) [inline transports/tls/CHANGELOG.md]:
```suggestion
# 0.1.0-alpha
```

**Review by thomaseizinger — DISMISSED** [review]

**Comment by thomaseizinger** [comment]
Mergify can't update your branch to the latest version of master: https://github.com/libp2p/rust-libp2p/pull/3196/checks?check_run_id=10175104797

Can you see if you can address this? I've heard that it is tricky to do for forks coming from organisations but haven't confirmed that. Worst case, can you open this PR from a personal fork instead?

**Comment by ppodolsky** [comment]
> Mergify can't update your branch to the latest version of master: https://github.com/libp2p/rust-libp2p/pull/3196/checks?check_run_id=10175104797
> 
> Can you see if you can address this? I've heard that it is tricky to do for forks coming from organisations but haven't confirmed that. Worst case, can you open this PR from a personal fork instead?

Done, but had to force push rebased commits. 
I believe organisations' forks are ok if the organisation belongs to you.

**Comment by thomaseizinger** [comment]
> > Mergify can't update your branch to the latest version of master: [#3196 (checks)](https://github.com/libp2p/rust-libp2p/pull/3196/checks?check_run_id=10175104797)
> > Can you see if you can address this? I've heard that it is tricky to do for forks coming from organisations but haven't confirmed that. Worst case, can you open this PR from a personal fork instead?
> 
> Done, but had to force push rebased commits. I believe organisations' forks are ok if the organisation belongs to you.

Cool thanks! Let's see if mergify is happy now :)

**Review by thomaseizinger — DISMISSED** [review]

**Review by mxinden — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Some of the dependency bumps in this PR seem to be incorrect: https://github.com/libp2p/rust-libp2p/actions/runs/3729641763/jobs/6327231245

**Comment by ppodolsky** [comment]
@mxinden Seems something from your commit. Not sure what to do with it

**Comment by mxinden** [comment]
Should be fixed with https://github.com/libp2p/rust-libp2p/pull/3261 merged. Can you update the pull request @ppodolsky?

**Comment by mxinden** [comment]
> > > Mergify can't update your branch to the latest version of master: [#3196 (checks)](https://github.com/libp2p/rust-libp2p/pull/3196/checks?check_run_id=10175104797)
> > > Can you see if you can address this? I've heard that it is tricky to do for forks coming from organisations but haven't confirmed that. Worst case, can you open this PR from a personal fork instead?
> > 
> > 
> > Done, but had to force push rebased commits. I believe organisations' forks are ok if the organisation belongs to you.
> 
> Cool thanks! Let's see if mergify is happy now :)

Following up on this, for future pull requests, us being able to merge `master` ourselfs into your pull request would make both of our lifes a bit easier.

**Comment by mxinden** [comment]
:pray:

**Comment by ppodolsky** [comment]
> Following up on this, for future pull requests, us being able to merge master ourselfs into your pull request would make both of our lifes a bit easier.

Would be glad to allow but have no idea how to do it

**Comment by ppodolsky** [comment]
Now it is clippy(beta) failing

**Comment by mxinden** [comment]
> Now it is clippy(beta) failing

That CI check is not required, i.e. can be fixed in a different pull request. Only the stable clippy check is required.

**Comment by mxinden** [comment]
:tada: Thanks for the help here.

**Pull request merged**